scripts/database.py

```diff
@@ -0,0 +1,182 @@
+                       r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|'  # domain...
+                       r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'  # ...or ip
+                       r'(?::\d+)?'  # optional port
+                       r'(?:/?|[/?]\S+)$', re.IGNORECASE)
+
+
+def validate_url(url):
+    if len(url) == 0:
+        return True
+    return url_regex.match(url) is not None
+
+
+def get_ban_list():
+    all_user_info = THE_DB.all()
+    tor = []
+    for user in all_user_info:
+        if user['id'] != 'admin':
+            banned = False
+            if 'banned' in user:
+                banned = user['banned']
+            tor.append((user['id'], banned))
+
+    return tor
+
+
+def make_user(username):
+    check_query = Query()
+    check_if_user_exists = THE_DB.search(check_query.id == username)
+    if (len(check_if_user_exists) > 0):
+        return
+    password = "%032x" % random.getrandbits(128)
+    new_user = {
+        'id': username,
+        'password': password,
+        'banned': False,
+        'dj_name': username,
+        'dj_pic': '',
+        'stream_title': '',
+        'stream_desc': '',
+    }
+
+    THE_DB.insert(new_user)
+    return (username, password)
+
+
+def get_dj_info(username):
+    check_query = Query()
+    check_if_user_exists = THE_DB.search(check_query.id == username)
+    if (len(check_if_user_exists) == 0):
+        return
+    return check_if_user_exists[0]
+
+
+def update_dj_info(username, form_dict):
+    check_query = Query()
+    check_if_user_exists = THE_DB.search(check_query.id == username)
+    if (len(check_if_user_exists) == 0):
+        return False
+    # only put in proper picture urls
+    if 'dj_pic' in form_dict:
+        if not validate_url(form_dict['dj_pic']):
+            del form_dict['dj_pic']
+    THE_DB.update(form_dict, check_query.id == username)
+    return True
+
+
+def change_password(username, new_pass=None):
+    check_query = Query()
+    check_if_user_exists = THE_DB.search(check_query.id == username)
+    if (len(check_if_user_exists) == 0):
+        return
+    if new_pass is None:
+        new_pass = "%032x" % random.getrandbits(128)
+    THE_DB.update({'password': new_pass}, check_query.id == username)
+    return new_pass
+
+
+def ban_user(username, new_ban_status):
+    check_query = Query()
+    check_if_user_exists = THE_DB.search(check_query.id == username)
+    if (len(check_if_user_exists) == 0):
+        return
+    THE_DB.update({'banned': new_ban_status}, check_query.id == username)
+    return new_ban_status
+
+
+class DJUser(object):
+    def __init__(self, username, password):
+        self.id = username
+        self.password = password
+
+    @property
+    def is_active(self):
+        # check ban status
+        check_ban = Query()
+        check_ban_res = THE_DB.search(check_ban.id == self.id)
+        if len(check_ban_res) == 0:
+            return False
+        if 'banned' not in check_ban_res[0]:
```

Comment:
I feel this should be a boolean, rather than a substring comparison.  A user could just have an `is_active` flag, rather than this more general notion of banning.